src/main/python/utils/s3_util_class.py

```diff
@@ -0,0 +1,84 @@
+import csv
+import boto3 as bt
+import pandas as pd
+import json
+import uuid
+
+
+class Uploader:
+    # not sure about initilization info yet
+    # I assumed different data objects might have different bucket names/keys
+    # otherwise we could initialize the class with a unique bucket_name and key
+    # for all data input
+
+    # create unique file names to minimize collisions in storage
+    def create_unique_name(self, data_object):
+        file_info = ''.join([key for key in data_object.keys()])
+        random_id = uuid.uuid4().hex[:8]
+        return ''.join(str(random_id, file_info, ".csv"))
+
+
+    def dict_to_csv(self, dict_object):
+        # assuming the dict object is a dict of dicts
+        # create a unique id from all the first level keys
+        file_name = self.create_unique_name(dict_object)
+
+        open_csv = open(file_name, 'w')
+        csv_writer = csv.writer(open_csv)
+
+        for item in dict_object:
+            csv_writer.writerow(dict_object[item])
+
+        return file_name
+
+
+    def json_to_csv(self, json_object):
+        json_data = json.loads(json_object)
+
+        # create unique id from all the keys of first json element
+        file_name = self.create_unique_name(json_data[0])
+
+        open_csv = open(file_name, 'w')
+        csv_writer = csv.writer(open_csv)
+
+        for item in json_data:
+            csv_writer.writerow(json.dumps(item))
+
+        return file_name
+
+
+    def panda_df_to_csv(self, panda_df):
+        # first convert panda data frame to traditional json format
+        json_object = panda_df.to_json(orient='records')
+
+        # execute json to csv routine on new json data
+        file_name = self.json_to_csv(json_object)
+
+        return file_name
```

Comment:
Above three methods and last three can be merged if feel. The logic is same.?

Reply:
We can have a conditional to check object type and run the loop? Also close the file after writing

Reply:
On it.